types/pdfmake/interfaces.d.ts

```diff
@@ -0,0 +1,473 @@
+    paddingRight?: number | DynamicLayout<number>;
+    paddingTop?: DynamicLayout<number>;
+    paddingBottom?: DynamicLayout<number>;
+    fillOpacity?: number | DynamicLayout<number>;
+    defaultBorder?: boolean;
+}
+
+export type DynamicLayout<T> = (rowIndex: number, node: ContentTable, columnIndex: number) => T | null | undefined;
+
+export interface LineStyle {
+    dash?: {
+        length: number;
+        space?: number;
+    };
+}
+
+export type TableCell =
+    | {} // Used when another cell spans over this cell
+    | (Content & {
+          rowSpan?: number;
+          colSpan?: number;
+          border?: [boolean, boolean, boolean, boolean];
+          borderColor?: [string, string, string, string];
+          fillOpacity?: number;
+      });
+
+export interface Table {
+    body: TableCell[][];
+    widths?: Size[];
+    heights?: number | number[] | DynamicRowSize;
+    headerRows?: number;
+    dontBreakRows?: boolean;
+    keepWithHeaderRows?: number;
+    layout?: TableLayout;
+}
+
+export type PredefinedTableLayout = 'noBorders' | 'headerLineOnly' | 'lightHorizontalLines';
+export type TableLayout = PredefinedTableLayout | CustomTableLayout;
+
+export interface Style {
+    font?: string;
+    fontSize?: number;
+    fontFeatures?: PDFKit.Mixins.OpenTypeFeatures[];
+    lineHeight?: number;
+    bold?: boolean;
+    italics?: boolean;
+    alignment?: Alignment;
+    color?: string;
+    background?: string;
+    markerColor?: string;
+    decoration?: Decoration;
+    decorationStyle?: DecorationStyle;
+    decorationColor?: string;
+    margin?: Margins;
+    preserveLeadingSpaces?: boolean;
+    opacity?: number;
+    characterSpacing?: number;
+    leadingIndent?: number;
+    // Table-cell properties:
+    noWrap?: boolean;
+    fillColor?: string;
+    fillOpacity?: number;
+    columnGap?: Size;
+    // These properties appear in the documentation but don't do anything:
+    // tableCellPadding?: unknown;
+    // cellBorder?: unknown;
+    // headerCellBorder?: unknown;
+    // oddRowCellBorder?: unknown;
+    // evenRowCellBorder?: unknown;
+    // tableBorder?: unknown;
+}
+
+export type Content =
+    | string
+    | ArrayOfContent
+    | ContentText
+    | ContentColumns
+    | ContentStack
+    | ContentUnorderedList
+    | ContentOrderedList
+    | ContentTable
+    | ContentAnchor
+    | ContentPageReference
+    | ContentTextReference
+    | ContentToc
+    | ContentTocItem
+    | ContentImage
+    | ContentSvg
+    | ContentQr
+    | ContentCanvas;
+
+// not exported, only used to prevent Content from circularly referencing itself
+interface ArrayOfContent extends Array<Content> {}
+
+export interface ContentText extends ContentLink, ContentBase {
+    text: Content;
+}
+
+export interface ContentColumns extends ContentBase {
+    columns: Column[];
+}
+
+export interface ContentStack extends ContentBase {
+    stack: Content[];
+}
+
+export interface ContentOrderedList extends ContentBase {
```

Comment:
Please add back type annotation comments.

Reply:
I didn't pay attention while merging, will do.

Reply:
This looks good.